### PR TITLE
Author datetime fix

### PR DESCRIPTION
**Pull request by jbradl11**
Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

DON'T MERGE UNTIL ALL DEPENDENT PRS ARE MERGED.

This PR adds the author datetime field value to the necessary Datatypes. If a data criteria has an author datetime field value and is then negated, the field value author datetime will be removed and bonnie will default to the 'start time' author datetime.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1949
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass (regressions run, found, and documented)

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: spec/javascripts/fixtures/json/measure_data/special_measures/CMS52v7/CMS52v7.json [hunk 1]
```diff
@@ -27178,7 +27184,7 @@
   "title": "HIV/AIDS: Pneumocystis Jiroveci Pneumonia (PCP) Prophylaxis",
   "type": "ep",
   "updated_at": "2018-04-02T18:37:23.834Z",
-  "user_id": "5a57e975942c6d1e61d32f13",
+  "user_id": null,
```
Comment (lukoktonos) [inline spec/javascripts/fixtures/json/measure_data/special_measures/CMS52v7/CMS52v7.json:27187]: Why did all these `user_id`s get nulled out?

Comment (hossenlopp) [reply spec/javascripts/fixtures/json/measure_data/special_measures/CMS52v7/CMS52v7.json:27187]: I think it was a recent-ish change in the export script that doesn't quite get the user_id setting correct. To be honest, the user_ids really don't matter for frontend fixtures.

File: spec/javascripts/views/patient_builder_spec.js.coffee [hunk 2]
```diff
@@ -180,6 +180,20 @@ describe 'PatientBuilderView', ->
...
+      authorDateTimePatient = @patients.models.filter((patient) -> patient.get('last') is 'AuthorDateTime')[0]
+      patientBuilder = new Thorax.Views.PatientBuilder(model: authorDateTimePatient, measure: @measure, patients: @patients)
+      patientBuilder.appendTo 'body'
+      firstCriteria = patientBuilder.model.get('source_data_criteria').first()
+      authorDateTime = patientBuilder.model.get('source_data_criteria').first().get('field_values').first().get('start_date')
+      expect(authorDateTime).toEqual '04/24/2019'
+      patientBuilder.$('input[name=negation]:first').click()
+      expect(patientBuilder.model.get('source_data_criteria').first().get('field_values').length).toBe 0
+      startDate = new Date(patientBuilder.model.get('source_data_criteria').first().get('start_date'))
+      expect(startDate.toString()).toEqual('Tue Apr 17 2012 04:00:00 GMT-0400 (Eastern Daylight Time)')
```
Comment (hossenlopp) [inline spec/javascripts/views/patient_builder_spec.js.coffee]: This makes the assumption that the user's browser is in EDT. This test fails in travis as a result.

File: spec/javascripts/views/patient_builder_spec.js.coffee [hunk 3]
```diff
@@ -180,6 +180,20 @@ describe 'PatientBuilderView', ->
 
     afterEach -> @patientBuilder.remove()
 
+  describe 'author date time', ->
+    it "removes author date time field value when not performed is checked", ->
+      authorDateTimePatient = @patients.models.filter((patient) -> patient.get('last') is 'AuthorDateTime')[0]
+      patientBuilder = new Thorax.Views.PatientBuilder(model: authorDateTimePatient, measure: @measure, patients: @patients)
+      patientBuilder.appendTo 'body'
+      firstCriteria = patientBuilder.model.get('source_data_criteria').first()
+      authorDateTime = patientBuilder.model.get('source_data_criteria').first().get('field_values').first().get('start_date')
```
Comment (hossenlopp) [inline spec/javascripts/views/patient_builder_spec.js.coffee:189]: Shouldn't this be checking the existence of `author_datetime`?

**Review by hossenlopp — CHANGES_REQUESTED** [review]
The new test breaks on travis due to system timezone differences and doesn't quite seem to check what it says it checks.

File: spec/javascripts/views/patient_builder_spec.js.coffee (hunk 3 above)
Comment (jbradl11) [reply spec/javascripts/views/patient_builder_spec.js.coffee:189]: The author datetime field value attribute has a value called "start time"

**Review by lukoktonos — APPROVED** [review]
🎉

**Pull request merged**